mgmt_cli_test.py

```diff
@@ -602,6 +603,53 @@ def test_basic_backup(self, ks_names: list = None):
         mgr_cluster.delete()  # remove cluster at the end of the test
         self.log.info('finishing test_basic_backup')
 
+    def test_no_delta_backup_at_disabled_compaction(self):
+        """The purpose of test is to check that delta backup (no changes to DB between backups) takes time -> 0.
+
+        Important test precondition is to disable compaction on all nodes in the cluster.
+        Otherwise, new set of SSTables is created what ends up in the situation that almost no deduplication is applied.
+
+        For more details https://github.com/scylladb/scylla-manager/issues/3936#issuecomment-2277611709
+        """
+        self.log.info('starting test_consecutive_backups')
+
+        self.log.info('Run write stress')
+        self.run_prepare_write_cmd()
+
+        self.log.info('Disable compaction for every node in the cluster')
+        compaction_ops = CompactionOps(cluster=self.db_cluster)
+        for node in self.db_cluster.nodes:
+            compaction_ops.disable_autocompaction_on_ks_cf(node=node)
```

Comment:
I'm not sure if disabling autocompaction stops ongoing compactions, or does it just prevent new ones from starting. If it does not stop them, then it's possible that there would be a compaction running during the backup. It would result in creating new files that needs to be uploaded as a part of the second backup.

You could create table, disable autocompaction and only after that run the c-s command to fill the table.

Reply:
I suppose we are pretty save in this regard. 
Here is the output of `sctool progress ...` for the two consecutive backup operations:
```
Running command "sudo sctool  -c 75f34691-49ca-4fdb-bfa3-d56229b4e809 progress backup/2a9c1120-6074-474b-80e5-001393f359f7" finished with status 0
Run:		96414835-5676-11ef-ac67-02a275dbf199
Status:		DONE
Start time:	09 Aug 24 17:41:19 UTC
End time:	09 Aug 24 17:43:39 UTC
Duration:	2m19s
Progress:	100%
Snapshot Tag:	sm_20240809174120UTC
Datacenters:	
- us-east

╭─────────────┬──────────┬─────────┬─────────┬──────────────┬────────╮
│ Host        │ Progress │    Size │ Success │ Deduplicated │ Failed │
├─────────────┼──────────┼─────────┼─────────┼──────────────┼────────┤
│ 10.12.0.64  │     100% │ 11.406G │ 11.406G │            0 │      0 │
│ 10.12.0.69  │     100% │ 11.406G │ 11.406G │            0 │      0 │
│ 10.12.2.186 │     100% │ 11.406G │ 11.406G │            0 │      0 │
╰─────────────┴──────────┴─────────┴─────────┴──────────────┴────────╯
```
No-delta backup operation (the test was executed from the PR branch):
```
Running command "sudo sctool  -c 75f34691-49ca-4fdb-bfa3-d56229b4e809 progress backup/783ae129-6762-46cb-98bb-4321aaf1d1e6"...

Run:		1045364a-5677-11ef-ac68-02a275dbf199
Status:		DONE
Start time:	09 Aug 24 17:44:44 UTC
End time:	09 Aug 24 17:44:47 UTC
Duration:	2s
Progress:	100%
Snapshot Tag:	sm_20240809174444UTC
Datacenters:	
- us-east

╭─────────────┬──────────┬─────────┬─────────┬──────────────┬────────╮
│ Host        │ Progress │    Size │ Success │ Deduplicated │ Failed │
├─────────────┼──────────┼─────────┼─────────┼──────────────┼────────┤
│ 10.12.0.64  │     100% │ 11.406G │ 11.406G │      11.406G │      0 │
│ 10.12.0.69  │     100% │ 11.406G │ 11.406G │      11.406G │      0 │
│ 10.12.2.186 │     100% │ 11.406G │ 11.406G │      11.406G │      0 │
╰─────────────┴──────────┴─────────┴─────────┴──────────────┴────────╯
```
As you can see, all the data is deduplicated in case of second backup and no size changes happened.

Reply:
If this is a problem, we have something implemented in perf tests `_wait_no_compactions_running` - could be moved and reused if necessary.

Reply:
in dtest when disabling compactions, we wait for compactions to finish before moving on.

even that it's not completely clear if backup can work as expected during compactions, why manager is stop the compactions on it's own on the tables is currently working on ?, why does the user need to care about it ?

Reply:
> in dtest when disabling compactions, we wait for compactions to finish before moving on.

I'll add this check in the test to wait compaction to finish.

> even that it's not completely clear if backup can work as expected during compactions, why manager is stop the compactions on it's own on the tables is currently working on ?, why does the user need to care about it ?

The test checks the specific case of data deduplication for consecutive no-delta backup. This scenario might be achieved only with compaction disabled. See more details in the [comment](https://github.com/scylladb/scylla-manager/issues/3936#issuecomment-2277611709).

cc @karol-kokoszka